sdk/iot/Azure.Iot.Hub.Service/src/Authentication/IotHubSasCredential.cs

```diff
@@ -0,0 +1,138 @@
+            var sharedAccessKey = iotHubConnectionString.GetRequired(SharedAccessSignatureConstants.SharedAccessKeyIdentifier);
+
+            Endpoint = BuildEndpointUriFromHostName(iotHubConnectionString.GetRequired(SharedAccessSignatureConstants.HostNameIdentifier));
+            SetCredentials(sharedAccessPolicy, sharedAccessKey);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="IotHubSasCredential"/> class.
+        /// </summary>
+        /// <param name="sharedAccessPolicy">
+        /// The IoT Hub access permission, which can be either "iothubowner", "service", "registryRead" or "registryReadWrite" policy, as applicable.
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#access-control-and-permissions"/>.
+        /// </param>
+        /// <param name="sharedAccessKey">
+        /// The IoT Hub shared access key associated with the shared access policy permissions.
+        /// </param>
+        /// <param name="timeToLive">
+        /// (Optional) The validity duration of the generated shared access signature token used for authentication.
+        /// The token will be renewed when at 15% or less of it's lifespan. The default value is 30 minutes.
+        /// </param>
+        public IotHubSasCredential(string sharedAccessPolicy, string sharedAccessKey, TimeSpan timeToLive = default)
+        {
+            SetCredentials(sharedAccessPolicy, sharedAccessKey, timeToLive);
+        }
+
+        private void SetCredentials(string sharedAccessPolicy, string sharedAccessKey, TimeSpan timeToLive = default)
+        {
+            SharedAccessPolicy = sharedAccessPolicy;
+            SharedAccessKey = sharedAccessKey;
+
+            if (timeToLive != null)
+            {
+                if (timeToLive.CompareTo(TimeSpan.Zero) < 0)
+                {
+                    throw new ArgumentException("The value for SasTokenTimeToLive cannot be a negative TimeSpan", nameof(timeToLive));
+                }
+
+                SasTokenTimeToLive = timeToLive;
+            }
+
+            _cachedSasToken = null;
+        }
+
+        /// <summary>
+        /// The IoT Hub service instance endpoint to connect to.
+        /// </summary>
+        public Uri Endpoint { get; internal set; }
+
+        /// <summary>
+        /// The IoT Hub access permission, which can be either "iothubowner", "service", "registryRead" or "registryReadWrite" policy, as applicable.
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#access-control-and-permissions"/>.
+        /// </summary>
+        public string SharedAccessPolicy { get; private set; }
+
+        /// <summary>
+        /// The IoT Hub shared access key associated with the shared access policy permissions.
+        /// </summary>
+        public string SharedAccessKey { get; private set; }
```

Comment:
This is currently settable only via the constructor. If we want to support rolling of shared access keys, then we will have to expose a thread-safe setter for this.